src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1123,14 +1159,32 @@ void primitive_inst::do_runtime_skip_permute() {
     }
     // If the largest value and total size are different, can_be_optimized needs to be reset
     if (size != max_value) {
-        GPU_DEBUG_TRACE_DETAIL << "--- Cannot optimize because size(" << size << ") and max_value(" << max_value << ") are different" << std::endl;
-        set_can_be_optimized(false);
+        GPU_DEBUG_TRACE_DETAIL << "--- Cannot optimize because size(" << size << ") and max_value(" << max_value
+                               << ") are different" << std::endl;
+        if (get_node().is_type<permute>())
+            set_can_be_optimized(false);
+        else
+            permute_user->set_can_be_optimized(false);
         return;
     }
-    GPU_DEBUG_TRACE_DETAIL << "[do_runtime_skip_permute] " << id() << " : can_be_optimized" << std::endl;
-    GPU_DEBUG_TRACE_DETAIL << "            - Input layout : " << _impl_params->get_input_layout(0).to_short_string() << std::endl;
-    GPU_DEBUG_TRACE_DETAIL << "            - Output layout : " << _impl_params->get_output_layout().to_short_string() << std::endl;
-    set_can_be_optimized(true);
+
+    if (get_node().is_type<permute>()) {
+        GPU_DEBUG_TRACE_DETAIL << "[do_runtime_skip_permute] " << id() << " : can_be_optimized" << std::endl;
+        GPU_DEBUG_TRACE_DETAIL << "            - Input layout : " << _impl_params->get_input_layout(0).to_short_string() << std::endl;
+        GPU_DEBUG_TRACE_DETAIL << "            - Output layout : " << _impl_params->get_output_layout().to_short_string() << std::endl;
+        set_can_be_optimized(true);
+    } else {
+        GPU_DEBUG_TRACE_DETAIL << "[prepare_runtime_skip_permute], user of " << id() << " : can_be_optimized" << std::endl;
+        GPU_DEBUG_TRACE_DETAIL << "            - Input layout : " << _impl_params->get_output_layout().to_short_string() << std::endl;
+        GPU_DEBUG_TRACE_DETAIL << "            - Output layout : "
+            << permute_user->get_node().get_kernel_impl_params()->get_output_layout().to_short_string() << std::endl;
+        permute_user->set_can_be_optimized(true);
+        // needs to add extra syncronization
+        // permute as user can be optimized, check pattern permute + reorder, which may be both optimized during runtime
+        _needs_completion_event |= permute_user->needs_completion_event();
```

Comment:
@yeonbok taylor, can  you help check if this method is applicable by adding extra sync point?